DEPRECATED.md

```diff
@@ -15,7 +15,36 @@ You can use the static helper methods such as `System::loadLanguageFile()` or
 The constants `TL_ROOT`, `TL_MODE`, `TL_START` and `TL_SCRIPT` have been
 deprecated and will be removed in Contao 5.0.
 
-// FIXME: show the alternatives
+Instead of `TL_ROOT` use:
+
+```php
+global $kernel;
+
+$rootDir = $kernel->getContainer()->getParameter('kernel.root_dir');
+```
+
+Instead of `TL_MODE` to check for `BE` and `FE` use:
+
+```php
+global $kernel;
+
+$isFE = $kernel->getContainer()->isScopeActive('frontend');
+$isBE = $kernel->getContainer()->isScopeActive('backend');
+```
+
+Note: You can also use the `ScopeAwareListener` in your own event listeners.
+
+Instead of `TL_START` use:
+```php
+global $kernel;
+
+$startTime = $kernel->getStartTime();
+```
+
+Note: This will only work if Symfony is in debug mode (`kernel.debug`).
+
+Instead of `TL_SCRIPT` use the Symfony routing component to generate proper URLs.
```

Comment:
Aren't people mainly using `TL_SCRIPT` to check for the current entry point?

```php
if (TL_SCRIPT == 'contao/main.php') {
}
```

We should probably describe both cases, shouldn't we?